modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/consensus/ConsensusStateUpdater.scala

```diff
@@ -0,0 +1,182 @@
+package org.tessellation.sdk.infrastructure.consensus
+
+import java.security.KeyPair
+
+import cats.data.NonEmptyList
+import cats.effect.Async
+import cats.kernel.Next
+import cats.syntax.all.none
+import cats.syntax.flatMap._
+import cats.syntax.foldable._
+import cats.syntax.functor._
+import cats.syntax.option._
+import cats.syntax.show._
+import cats.syntax.traverse._
+import cats.{Applicative, Order, Show}
+
+import scala.reflect.ClassTag
+import scala.reflect.runtime.universe.TypeTag
+
+import org.tessellation.ext.crypto._
+import org.tessellation.kryo.KryoSerializer
+import org.tessellation.schema.node.NodeState
+import org.tessellation.schema.peer.PeerId
+import org.tessellation.sdk.domain.cluster.storage.ClusterStorage
+import org.tessellation.sdk.domain.consensus.ConsensusFunctions
+import org.tessellation.sdk.domain.gossip.Gossip
+import org.tessellation.sdk.infrastructure.consensus.message._
+import org.tessellation.security.SecurityProvider
+import org.tessellation.security.hash.Hash
+import org.tessellation.security.signature.Signed
+import org.tessellation.security.signature.signature.{Signature, SignatureProof}
+
+import org.typelevel.log4cats.slf4j.Slf4jLogger
+
+trait ConsensusStateUpdater[F[_], Key, Artifact] {
+
+  def tryAdvanceConsensus(
+    key: Key,
+    resources: ConsensusResources[Artifact]
+  ): F[Option[ConsensusState[Key, Artifact]]]
+
+  def tryFacilitateConsensus(
+    key: Key,
+    lastKeyAndArtifact: (Key, Artifact)
+  ): F[Option[ConsensusState[Key, Artifact]]]
+}
+
+object ConsensusStateUpdater {
+
+  def make[F[_]: Async: KryoSerializer: SecurityProvider, Event, Key: Show: Order: Next: TypeTag: ClassTag, Artifact <: AnyRef: Show: TypeTag](
+    consensusFns: ConsensusFunctions[F, Event, Key, Artifact],
+    consensusStorage: ConsensusStorage[F, Event, Key, Artifact],
+    clusterStorage: ClusterStorage[F],
+    gossip: Gossip[F],
+    keyPair: KeyPair,
+    selfId: PeerId
+  ): ConsensusStateUpdater[F, Key, Artifact] = new ConsensusStateUpdater[F, Key, Artifact] {
+
+    private val logger = Slf4jLogger.getLoggerFromClass(ConsensusStateUpdater.getClass)
+
+    def tryFacilitateConsensus(
+      key: Key,
+      lastKeyAndArtifact: (Key, Artifact)
+    ): F[Option[ConsensusState[Key, Artifact]]] =
+      consensusStorage
+        .updateState(key)(logIfUpdated(internalTryFacilitateConsensus(key, lastKeyAndArtifact)))
```

Comment:
It seems like we are duplicating logs here. We have a log here and inside of the storage for the same case - success. I guess we could remove one of them.